diff --git a/c/Makefile.am b/c/Makefile.am
index 29a1e875..f6ecc180 100644
--- a/c/Makefile.am
+++ b/c/Makefile.am
@@ -19,7 +19,7 @@ mlr_LDADD=		\
 			lib/libmlr.la \
 			parsing/libdsl.la \
 			aux/libaux.la \
-			-lm
+			-lm -lpcreposix
 
 # Other executable variants
 
diff --git a/c/lib/mlr_arch.h b/c/lib/mlr_arch.h
index 97eaf5c8..30fe1d0c 100644
--- a/c/lib/mlr_arch.h
+++ b/c/lib/mlr_arch.h
@@ -4,6 +4,8 @@
 #include <stdio.h>
 #include <time.h>
 
+#define MLR_ON_MSYS2
+
 // ----------------------------------------------------------------
 // Miller is single-threaded and the file-locking in getc is simply an unneeded
 // performance hit, so we intentionally call getc_unlocked().  But for MSYS2
